django/http/response.py

```diff
@@ -383,7 +384,10 @@ def streaming_content(self):
     @streaming_content.setter
     def streaming_content(self, value):
         # Ensure we can never iterate on "value" more than once.
-        self._iterator = iter(value)
+        if hasattr(value, 'read'):
```

Comment:
I would go with the "could" option and port `FileResponse` to 1.7 rather than change an existing (stable) class.